sympy/physics/control/lti.py

```diff
@@ -0,0 +1,1033 @@
+                            (self.represent[2].shape[0] == self.represent[3].shape[0])):
+                        raise ShapeError("Shapes of A,B,C,D must fit")
+
+    def find_realization(self, G, s):
+        """ Representation (A, B, C, D) of the state space model
+
+        Returns the representation in state space of a given transfer function
+
+        Parameters
+        ==========
+
+        G: Matrix
+            Matrix valued transfer function G(s) in Laplace space
+        s: Symbol
+            variable s, where G is dependent from
+
+        See Also
+        ========
+
+        Utils : some quick tools for matrix polynomials
+
+        References
+        ==========
+
+        Joao P. Hespanha, Linear Systems Theory. 2009.
+        """
+
+        A, B, C, D = 4 * [None]
+
+        try:
+            m, k = G.shape
+
+        except AttributeError:
+            raise TypeError("G must be a matrix.")
+
+        # test if G is proper
+        if not is_proper(G, s, strict=False):
+            raise ValueError("G must be proper.")
+
+        # define D as the limit of G for s to infinity
+        D = G.limit(s, oo)
+
+        # define G_sp as the (stricly proper) difference of G and D
+        G_sp = G - D
+
+        # get the coefficients of the monic least common denominator of all entries of G_sp
+        # compute a least common denominator using utl and lcm
+        lcd = lcm(list(_fraction_list(G_sp, only_denoms=True)))
+
+        # make it monic
+        lcd = lcd / LC(lcd, s)
+
+        # and get a coefficient list of its monic. The [1:] cuts the LC away (thats a one)
+        lcd_coeff = Poly(lcd, s).all_coeffs()[1:]
+
+        # get the degree of the lcd
+        lcd_deg = degree(lcd, s)
+
+        # get the Matrix Valued Coeffs of G_sp in G_sp = 1/lcd * (N_1 * s**(n-1) + N_2 * s**(n-2) .. + N_n)
+        G_sp_coeff = _matrix_coeff(simplify(G_sp * lcd), s)
+        G_sp_coeff = [zeros(m, k)] * (lcd_deg - len(G_sp_coeff)) + G_sp_coeff
+
+        # now store A, B, C, D in terms of the coefficients of lcd and G_sp
+        # define A
+        A = (-1) * lcd_coeff[0] * eye(k)
+
+        for alpha in lcd_coeff[1:]:
+            A = A.row_join((-1) * alpha * eye(k))
+
+        for i in range(lcd_deg - 1):
+            tmp = eye(k) if i == 0 else zeros(k)
+
+            for j in range(1, lcd_deg):
+                tmp = tmp.row_join(eye(k)) if j == i else tmp.row_join(zeros(k))
+            if tmp:
+                A = A.col_join(tmp)
+
+        # define B
+        B = eye(k)
+        for i in range(lcd_deg - 1):
+            B = B.col_join(zeros(k))
+
+        # define C
+        C = G_sp_coeff[0]
+        for i in range(1, lcd_deg):
+            C = C.row_join(G_sp_coeff[i])
+
+        # return the state space representation
+        return (A, B, C, D)
+
+    #
+    # evaluate(self, u, t)
+    #
+    def evaluate(self, u, x0, t, t0=0, **flags):
+        """evaluate the system output for an input u
```

Comment:
You need to read through the docstring style guide: https://docs.sympy.org/latest/documentation-style-guide.html and make sure your docstrings follow the guide. There are numerous details to correct in this PR.